repack.sh

```diff
@@ -106,37 +159,42 @@ repackinf()
 	elif echo "${FIRMWARE}" | grep -q ME; then
 		CATEGORY="X-ManagementEngine"
 	fi
-	sed -i "s|{CATEGORY}|${CATEGORY}|g" "${TEMP}/firmware.metainfo.xml"
 
 	# Update firmware version
 	VERSION="$(grep 'FirmwareVersion' "${TEMP}/firmware.inf" | cut -d',' -f5 | sed 's|\r||')"
 	MAJOR="$(( (VERSION >> 24) & 0xff ))"
 	MINOR="$(( (VERSION >> 16) & 0xff ))"
 	REV="$(( VERSION & 0xffff ))"
 	VERSION="${MAJOR}.${MINOR}.${REV}"
-	sed -i "s|{VERSION}|${VERSION}|g" "${TEMP}/firmware.metainfo.xml"
 
 	# Update firmware timestamp
 	TIMESTAMP="$(grep '^DriverVer' "${TEMP}/firmware.inf" | sed -E 's| +||g')"
 	TIMESTAMP="$(echo "${TIMESTAMP}" | cut -d'=' -f2 | cut -d',' -f1)"
 	TIMESTAMP="$(date '+%s' --date "${TIMESTAMP}")"
-	sed -i "s|{TIMESTAMP}|${TIMESTAMP}|g" "${TEMP}/firmware.metainfo.xml"
+
+	# Create metainfo file from $DEVICE, $CATEGORY, $VERSION, & $TIMESTAMP
+	filltemplate > "${TEMP}/firmware.metainfo.xml"
```

Comment:
I would prefer if these variables were passed as explicit parameters here. I don't know enough bash to definitly say what makes this work and what would break it.

Reply:
Sounds good. I'll fix that.